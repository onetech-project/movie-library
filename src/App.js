import 'react-native-gesture-handler';
import React from 'react';
import SplashScreen from 'react-native-splash-screen';
import { Provider } from 'react-redux';
import MainNavigation from './navigation/navigation';
import { store } from './stores';

class App extends React.Component {
  componentDidMount() {
    SplashScreen.hide();
  }

  render() {
    return (
      <>
        <Provider store={store}>
          <MainNavigation />
        </Provider>
      </>
    );
  }
}

export default App;
