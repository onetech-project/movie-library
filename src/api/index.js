import * as constant from './constant';
import * as authAPI from './auth';

export default {
  ...constant,
  ...authAPI,
};
