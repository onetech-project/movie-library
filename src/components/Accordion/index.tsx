import React, { useState } from 'react';
import { Text, View } from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';
import Icon from 'react-native-vector-icons/Ionicons';
import { Colors } from '../../utils';
import styles from './styles';

interface Props {
  title: string
}

const Accordion: React.FC<Props> = (props) => {
  const [show, setShow] = useState(false);
  const [caret, setCaret] = useState('forward');

  const handlePress = () => {
    setShow(!show);
    setCaret(show ? 'forward' : 'down');
  }

  return (
    <View style={styles.container}>
      <TouchableOpacity activeOpacity={1} style={styles.header} onPress={handlePress}>
        <Text style={styles.title}>{props.title}</Text>
        <Icon name={`ios-caret-${caret}`} size={25} color={Colors.red} />
      </TouchableOpacity>
      {show && (
        <>
          <View style={[styles.itemContainer, !props.children ? styles.contentEmpty : null, styles.list]}>
            {props.children || (
              <Icon size={25} name="ios-close-circle" color={Colors.gray70} />
            )}
          </View>
        </>
      )}
    </View>
  )
}

Accordion.defaultProps = {
  title: 'Accordion',
};

export default Accordion;