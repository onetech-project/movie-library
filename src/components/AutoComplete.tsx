import React, { useState, useEffect } from 'react';
import {
  View,
  Text,
  FlatList,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import Input, { InputProps } from './Input';
import { Colors } from '../utils';
import { connect } from 'react-redux'

interface Props extends InputProps {
  onPressItem?: (item: {}) => {},
  titleProperty: string,
  textProperty: string,
  apiUrl: string,
  login: any,
}

const renderItems = ({ item, onPress, titleProperty, textProperty }) => (
  <TouchableOpacity style={styles.AutoCompleteResultItem} onPress={() => onPress(item)}>
    <Text style={styles.titleText}>{item[titleProperty]}</Text>
    <Text style={styles.text}>{item[textProperty]}</Text>
  </TouchableOpacity>
);

const AutoComplete: React.FC<Props> = (props) => {
  const [input, setInput] = useState('');
  const [data, setData] = useState([]);

  useEffect(() => {
    const delayDebounceFn = setTimeout(async () => {
      if (input) {
        try {
          const result = await fetch(props.apiUrl).then(response => response.json());
          setData(result.slice(0, 10));
        } catch (error) {
          console.log(error);
          setData([]);
        }
      }
    }, 1000);

    return () => clearTimeout(delayDebounceFn)
  }, [input])

  return (
    <View>
      <Input
        placeholder={props.placeholder}
        value={input}
        onChangeText={setInput}
        keyboardType={props.keyboardType}
        returnKeyType={props.returnKeyType}
        placeholderTextColor={Colors.grayNearWhite}
      />
      {data && data.length > 0 && (
        <FlatList
          style={styles.list}
          data={data}
          contentContainerStyle={styles.AutoCompleteResultList}
          renderItem={({ item }) => renderItems({
            item,
            onPress: (item: {}) => {
              props.onPressItem(item);
              setInput('');
              setData([]);
            },
            titleProperty: props.titleProperty,
            textProperty: props.textProperty
          })}
          keyboardShouldPersistTaps="handled"
          nestedScrollEnabled
        />
      )}
    </View>
  )
};

const styles = StyleSheet.create({
  list: {
    zIndex: 1,
    position: 'absolute',
    marginTop: 65,
    width: '100%',
    maxHeight: 250,
  },
  AutoCompleteResultList: {
    backgroundColor: Colors.white,
    marginHorizontal: 10,
    borderWidth: 1,
    borderColor: Colors.whiteSmoke,
    borderBottomLeftRadius: 10,
    borderBottomRightRadius: 10,
    elevation: 1
  },
  AutoCompleteResultItem: {
    padding: 10,
    marginHorizontal: 10,
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderBottomColor: Colors.whiteSmoke,
  },
  titleText: {
    fontWeight: '600',
    color: Colors.f3f,
    marginTop: 5,
    marginBottom: 3,
    fontSize: 16
  },
  text: {
    marginLeft: 2,
    color: Colors.gray97
  }
});

AutoComplete.defaultProps = {
  placeholder: 'Search',
  returnKeyType: 'done',
  keyboardType: 'default'
};

const mapStateToProps = (state: any) => ({
  login: state.login,
})

export default connect(mapStateToProps, null)(AutoComplete);