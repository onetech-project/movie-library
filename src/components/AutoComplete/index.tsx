import React, { useState, useEffect } from 'react';
import {
  View,
  Text,
  FlatList,
  TouchableOpacity,
} from 'react-native';
import Input, { InputProps } from '../Input';
import Loading from '../Loading';
import { Colors, GlobalStyles } from '../../utils';
import { connect } from 'react-redux';
import styles from './styles';
import axios from 'axios';
import Icon from 'react-native-vector-icons/Ionicons';

interface Props extends InputProps {
  onPressItem?: (item: {}) => {},
  titleProperty: string,
  textProperty: string,
  apiUrl: string,
  login?: any,
  getSelected?: (selected: any[]) => []
}

const renderItems = ({ index, item, onPress, titleProperty, textProperty, onPressDeleteItem }) => (
  <TouchableOpacity
    key={index?.toString()}
    style={[styles.AutoCompleteResultItem, onPressDeleteItem ? styles.selectedList : null]}
    onPress={() => onPress(item)}
    activeOpacity={onPressDeleteItem ? 1 : 0.2}
  >
    <View style={{ flex: 11 }}>
      <Text style={styles.titleText}>{item[titleProperty]}</Text>
      <Text style={styles.text}>{item[textProperty]}</Text>
    </View>
    {onPressDeleteItem && (
      <Icon style={{ flex: 1, alignSelf: 'center' }} name="ios-close-circle" size={18} color={Colors.red} onPress={() => onPressDeleteItem(index)} />
    )}
  </TouchableOpacity>
);

const renderLoading = () => (
  <Loading size="large" />
)

const AutoComplete: React.FC<Props> = (props) => {
  const [input, setInput] = useState('');
  const [data, setData] = useState([]);
  const [loading, setLoading] = useState(false);
  const [selected, setSelected] = useState([]);

  useEffect(() => {
    const delayDebounceFn = setTimeout(async () => {
      if (input) {
        try {
          setLoading(true);
          const result = await axios.get(props.apiUrl);
          setData(result.data);
          setLoading(false);
        } catch (error) {
          console.log(error);
          setData([]);
          setLoading(false);
        }
      }
    }, 1000);

    return () => clearTimeout(delayDebounceFn)
  }, [input, props.apiUrl])

  useEffect(() => {
    props.getSelected(selected)
  }, [selected])

  return (
    <View style={GlobalStyles.fullWidth}>
      <Input
        iconName="ios-search"
        placeholder={props.placeholder}
        value={input}
        onChangeText={(text) => {
          setInput(text);
          props.onChangeText?.(text);
        }}
        keyboardType={props.keyboardType}
        returnKeyType={props.returnKeyType}
        placeholderTextColor={Colors.grayNearWhite}
      />
      {loading && (
        <View style={styles.loading}>
          {renderLoading()}
        </View>
      )}
      {data && data.length > 0 && (
        <FlatList
          style={styles.list}
          data={data}
          contentContainerStyle={styles.AutoCompleteResultList}
          renderItem={({ item, index }) => renderItems({
            index,
            item,
            onPress: (item: {}) => {
              props.onPressItem?.(item);
              setSelected([...selected, item])
              setInput('');
              setData([]);
            },
            titleProperty: props.titleProperty,
            textProperty: props.textProperty,
            onPressDeleteItem: null,
          })}
          keyboardShouldPersistTaps="handled"
          nestedScrollEnabled
        />
      )}
      {selected.map((item, index) => renderItems({
        index,
        item,
        onPress: () => { },
        titleProperty: props.titleProperty,
        textProperty: props.textProperty,
        onPressDeleteItem: (index: any) => setSelected(selected.filter((x, i) => i !== index))
      }))}
    </View>
  )
};

AutoComplete.defaultProps = {
  placeholder: 'Search',
  returnKeyType: 'done',
  keyboardType: 'default',
  getSelected: () => []
};

const mapStateToProps = (state: any) => ({
  login: state.login,
})

export default connect(mapStateToProps, null)(AutoComplete);