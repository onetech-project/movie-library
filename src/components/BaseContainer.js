import React from 'react';
import {
  SafeAreaView,
  StatusBar,
  StyleSheet,
} from 'react-native';
import { Colors } from '../utils';

const BaseContainer = ({ children }) => (
  <>
    <StatusBar translucent barStyle="dark-content" backgroundColor={Colors.white} />
    <SafeAreaView style={styles.SafeAreaView1} />
    <SafeAreaView style={styles.SafeAreaView2}>
      {children}
    </SafeAreaView>
  </>
);

const styles = StyleSheet.create({
  SafeAreaView1: { backgroundColor: Colors.white, flex: 0 },
  SafeAreaView2: { flex: 1, backgroundColor: Colors.white },
});

export default BaseContainer;
