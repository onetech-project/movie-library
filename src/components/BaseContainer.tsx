import React from 'react';
import {
  SafeAreaView,
  StatusBar,
  StyleSheet,
  StatusBarProps,
  ScrollView,
} from 'react-native';
import { Colors } from '../utils';

interface Props extends StatusBarProps { }

const BaseContainer: React.FC<Props> = (props) => (
  <>
    <StatusBar translucent barStyle="dark-content" backgroundColor={Colors.white} {...props} />
    <SafeAreaView style={styles.SafeAreaView1} />
    <SafeAreaView style={styles.SafeAreaView2}>
      <ScrollView
        nestedScrollEnabled={true}
        style={styles.container}
        contentContainerStyle={{ flexGrow: 1 }}
        keyboardShouldPersistTaps="handled">
        {props.children}
      </ScrollView>
    </SafeAreaView>
  </>
);

const styles = StyleSheet.create({
  SafeAreaView1: { backgroundColor: Colors.white, flex: 0 },
  SafeAreaView2: { flex: 1, backgroundColor: Colors.white },
  container: { flex: 1, backgroundColor: Colors.white, top: 30, borderWidth: 1 }
});

export default BaseContainer;
