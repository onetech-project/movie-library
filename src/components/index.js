import BaseContainer from './BaseContainer';
import Input from './Input';
import Loading from './Loading';

export { BaseContainer, Loading, Input };
