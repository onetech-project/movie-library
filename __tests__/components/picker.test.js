import * as React from 'react';
import renderer from 'react-test-renderer';
import { Picker } from '../../src/components';

describe('test Input Component', () => {
  it('should render Input component', () => {
    const tree = renderer.create(<Picker data={[]} textProperty="" titleProperty="" placeholder="testing" value="" />);
    expect(tree).toMatchSnapshot();
  });
});
