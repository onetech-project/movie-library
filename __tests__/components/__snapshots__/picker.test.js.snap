// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Input Component should render Input component 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "backgroundColor": "#ffffff",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <TextInput
        editable={false}
        iconName="ios-caret-down"
        placeholder="testing"
        placeholderTextColor="#dddddd"
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#ffffff",
            "borderBottomRightRadius": 0,
            "borderColor": "#f5f5f5",
            "borderRadius": 3,
            "borderRightWidth": 0,
            "borderTopRightRadius": 0,
            "borderWidth": 1,
            "color": "#000000",
            "elevation": 1,
            "flex": 8,
            "marginBottom": 10,
            "padding": 10,
          }
        }
        value=""
      />
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#dddddd",
              "fontSize": 20,
            },
            Object {
              "backgroundColor": "#ffffff",
              "borderBottomLeftRadius": 0,
              "borderColor": "#f5f5f5",
              "borderLeftWidth": 0,
              "borderRadius": 3,
              "borderTopLeftRadius": 0,
              "borderWidth": 1,
              "elevation": 1,
              "flex": 1,
              "marginBottom": 10,
              "paddingLeft": 0,
              "paddingTop": 15,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
